### PR TITLE
Add Clemson University scraper

### PR DESCRIPTION
**Pull request by jodizzle**
https://etpr.app.clemson.edu/repository/syllabus_public.php

Funny example of a "captcha", but otherwise a straightforward scraper.

### CODE REVIEW

File: osp_scraper/spiders/clemson.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                    'hops_from_seed': 1,
+                    'source_url': response.url,
+                    'source_anchor': '',
+                },
+                callback=get_searches
+            )
+
+
+        def get_searches(response):
+            semesters = response.css('select[name="semester_selected"] option')
```
Comment (HarrisonGregg) [inline osp_scraper/spiders/clemson.py]: I like using `extract_first` and then checking if it was successful, but if you're not going to, you could shorten your code by using `extract` with something like: `response.css('select[name="semester_selected"] option::attr(value)').extract()`.

File: osp_scraper/spiders/clemson.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            # Skip the first entry because it's an empty string.
+            disciplines = response.css('select[name="subj_selected"] option')[1:]
+            for sem, subj in itertools.product(semesters, disciplines):
```
Comment (HarrisonGregg) [inline osp_scraper/spiders/clemson.py]: Out of curiosity, why use `itertools.product` instead of just a nested for loop?

**Comment by jodizzle** [comment]
@HarrisonGregg I updated the looping.  It elongates the process a little bit, but I think it's still readable.

File: osp_scraper/spiders/clemson.py (hunk 2 above)
Comment (jodizzle) [inline osp_scraper/spiders/clemson.py]: I just feel like it's an appropriate use of Python's libraries.  Especially helps with keeping the number of indentation levels in check.

File: osp_scraper/spiders/clemson.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+from ..spiders.CustomSpider import CustomSpider
+
+class ClemsonSpider(CustomSpider):
+    name = "clemson"
+    allowed_domains = ["clemson.edu"]
+
+    def start_requests(self):
+        start_url = "https://etpr.app.clemson.edu/repository/syllabus_public.php"
+
+        def do_captcha(response):
```
Comment (jodizzle) [inline osp_scraper/spiders/clemson.py]: Ah, yes, that makes more sense.

File: osp_scraper/spiders/clemson.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                    'source_url': response.url,
+                    'source_anchor': '',
+                },
+                callback=get_searches
+            )
+
+
+        def get_searches(response):
+            semesters = response.css('select[name="semester_selected"] option')
+            sem_codes = semesters.css('option::attr(value)').extract()
```
Comment (jodizzle) [inline osp_scraper/spiders/clemson.py]: Sure, we can try that.

**Comment by jodizzle** [comment]
@HarrisonGregg I refactored.

**Pull request merged**